applications/asset_tracker_v2/src/events/cloud_module_event.h

```diff
@@ -115,13 +127,15 @@ struct cloud_module_event {
 	enum cloud_module_event_type type;
 
 	union {
-		/** Variable that contains a new configuration received from the cloud service. */
+		/** New configuration received from the cloud service. */
 		struct cloud_data_cfg config;
-		/** Variable that contains data that was attempted to be sent. Could be used
+		/** Cloud location data received from the cloud service. */
+		struct location_data cloud_location;
```

Comment:
This requires including `modem/location.h` which requires `CONFIG_LOCATION=y`. In `native_posix` configuration this won't be set as location module is turned off.

I guess we have two options. To flag this parameter and its usage with `CONFIG_LOCATION` or `CONFIG_LOCATION_MODULE`. Or create an own `cloud_location_data` structure with similar information to `location_data`.

@simensrostad Opinions?

Reply:
I'm not sure what I prefer. I guess flagging is fine as long as it doesn't spawm more flagging across modules to accommadate this.

Reply:
I've done the flagging to get native_posix compilation through so please check the instances of `#if defined(CONFIG_LOCATION_MODULE)` in this PR.